datafusion/physical-expr/src/analysis.rs

```diff
@@ -191,12 +191,15 @@ fn shrink_boundaries(
     })?;
     let final_result = graph.get_interval(*root_index);
 
+    // If during selectivity calculation we encounter an error, use 1.0 as cardinality estimate
+    // safest estimate(e.q largest possible value).
     let selectivity = calculate_selectivity(
         &final_result.lower.value,
         &final_result.upper.value,
         &target_boundaries,
         &initial_boundaries,
-    )?;
+    )
+    .unwrap_or(1.0);
```

Comment:
I think this is an example of  https://github.com/apache/arrow-datafusion/issues/7552 -- ignoring errors in the normal path results in potential slowdowns of planning

can we please fix the underlying issue here rather than ignoring it? Maybe we need to change the interval analysis API to return `Result<Option<>>` rather than just `Result` 

In general ignoring the errors I think is just papering over real problems

Reply:
I agree with you. I will remove this error ignoring, in the subsequent PR, where I will
- remove Statistics::default
- Remove `unwrap_or`, error ignoring.